client/my-sites/plans/current-plan/purchases-listing.jsx

```diff
@@ -0,0 +1,196 @@
+		purchase: PropTypes.object,
+	};
+
+	isLoading() {
+		const { selectedSite, isRequestingSitePlans: isRequestingPlans } = this.props;
+
+		return ! selectedSite || isRequestingPlans;
+	}
+
+	isFreePlan() {
+		const { currentPlan } = this.props;
+
+		return ! currentPlan || isFreePlan( currentPlan ) || isFreeJetpackPlan( currentPlan );
+	}
+
+	getPlanObject() {
+		const { selectedSite } = this.props;
+		const currentPlanSlug = get( selectedSite, 'plan.product_slug' );
+
+		if ( ! currentPlanSlug ) {
+			return null;
+		}
+
+		return getPlan( currentPlanSlug );
+	}
+
+	getActionButton() {
+		const { currentPlan, purchase, selectedSite, translate } = this.props;
+		const siteSlug = selectedSite ? selectedSite.slug : null;
+
+		if ( ! currentPlan || ! siteSlug ) {
+			return null;
+		}
+
+		// For free plan show a button redirecting to the plans comparison.
+		if ( this.isFreePlan() ) {
+			return <Button href={ `/plans/${ siteSlug }` }>{ translate( 'Compare Plans' ) }</Button>;
+		}
+
+		// Do not show the action button for partner plans.
+		const isPartnerPlan = purchase && isPartnerPurchase( purchase );
+		if ( isPartnerPlan ) {
+			return null;
+		}
+
+		// Show action button only to site owners.
+		if ( ! currentPlan.userIsOwner || ! currentPlan.id ) {
+			return null;
+		}
+
+		const hasAutoRenew = !! currentPlan.autoRenew;
+		let label = translate( 'Manage Plan' );
+		if (
+			! hasAutoRenew &&
+			! shouldAddPaymentSourceInsteadOfRenewingNow( currentPlan.expiryMoment )
+		) {
+			label = translate( 'Renew Now' );
+		}
+
+		return (
+			<Button href={ managePurchase( siteSlug, currentPlan.id ) } compact>
+				{ label }
+			</Button>
+		);
+	}
+
+	getPlanExpiration() {
+		const { currentPlan, purchase, translate } = this.props;
+		const isPartnerPlan = purchase && isPartnerPurchase( purchase );
+
+		if ( this.isFreePlan() || isPartnerPlan ) {
+			return null;
+		}
+
+		const hasAutoRenew = !! currentPlan.autoRenew;
+		if ( hasAutoRenew && currentPlan.autoRenewDateMoment ) {
+			return translate( 'Set to auto-renew on %s.', {
+				args: invoke( currentPlan, 'autoRenewDateMoment.format', 'LL' ),
+			} );
+		}
+
+		return translate( 'Expires on %s.', {
+			args: invoke( currentPlan, 'expiryMoment.format', 'LL' ),
+		} );
+	}
+
+	getPlanTitle() {
+		const planObject = this.getPlanObject();
+
+		if ( planObject.getTitle ) {
+			return planObject.getTitle();
+		}
+
+		return null;
+	}
+
+	getPlanTagLine() {
+		const { translate } = this.props;
+		const planObject = this.getPlanObject();
+
+		if ( planObject.getTagline ) {
+			return planObject.getTagline();
+		}
+
+		return translate(
+			'Unlock the full potential of your site with all the features included in your plan.'
+		);
+	}
+
+	renderPlan() {
+		const { isPlanExpiring, selectedSite, translate } = this.props;
+		const currentPlanSlug = get( selectedSite, 'plan.product_slug' );
+
+		return (
+			<Fragment>
+				<Card compact>
+					<strong>{ translate( 'My Plan' ) }</strong>
```

Comment:
I noticed that this creates some visual duplication. Is it expected @eeeeevon13? 

![](https://cldup.com/lk5uKdGYPX.png)